diff --git a/init/main.c b/init/main.c
--- a/init/main.c
+++ b/init/main.c
@@ -634,6 +634,19 @@ static void __init setup_command_line(char *command_line)
 {
 	size_t len, xlen = 0, ilen = 0;
 
+	{
+		char *ks;
+
+		while ((ks = strstr(command_line, "verifiedbootstate=orange")))
+			memcpy(ks, "verifiedbootstate=green ", 24);
+		while ((ks = strstr(command_line, "verifiedbootstate=red")))
+			memcpy(ks, "verifiedbootstate=green", 23);
+		while ((ks = strstr(command_line, "flash.locked=0")))
+			memcpy(ks, "flash.locked=1", 14);
+		while ((ks = strstr(command_line, "veritymode=logging")))
+			memcpy(ks, "veritymode=enforcing", 20);
+	}
+
 	if (extra_command_line)
 		xlen = strlen(extra_command_line);
 	if (extra_init_args)
diff --git a/fs/proc/cmdline.c b/fs/proc/cmdline.c
--- a/fs/proc/cmdline.c
+++ b/fs/proc/cmdline.c
@@ -3,14 +3,34 @@
 #include <linux/init.h>
 #include <linux/proc_fs.h>
 #include <linux/seq_file.h>
+#include <linux/slab.h>
+#include <linux/string.h>
 
 static int cmdline_proc_show(struct seq_file *m, void *v)
 {
-	seq_puts(m, saved_command_line);
-	seq_putc(m, '\n');
+	char *c = kstrdup(saved_command_line, GFP_KERNEL);
+
+	if (c) {
+		char *s;
+
+		while ((s = strstr(c, "verifiedbootstate=orange")))
+			memcpy(s, "verifiedbootstate=green ", 24);
+		while ((s = strstr(c, "verifiedbootstate=red")))
+			memcpy(s, "verifiedbootstate=green", 23);
+		while ((s = strstr(c, "flash.locked=0")))
+			memcpy(s, "flash.locked=1", 14);
+		while ((s = strstr(c, "veritymode=logging")))
+			memcpy(s, "veritymode=enforcing", 20);
+		seq_puts(m, c);
+		kfree(c);
+	} else {
+		seq_puts(m, saved_command_line);
+	}
+
+	seq_putc(m, '\n');
 	return 0;
 }
 
 static int __init proc_cmdline_init(void)
 {
 	proc_create_single("cmdline", 0, NULL, cmdline_proc_show);
diff --git a/fs/proc/bootconfig.c b/fs/proc/bootconfig.c
--- a/fs/proc/bootconfig.c
+++ b/fs/proc/bootconfig.c
@@ -6,17 +6,38 @@
 #include <linux/seq_file.h>
 #include <linux/bootconfig.h>
 #include <linux/bug.h>
+#include <linux/slab.h>
+#include <linux/string.h>
 #include <linux/kmemleak.h>
 #include "internal.h"
 
 static char *saved_boot_config;
 
 static int boot_config_proc_show(struct seq_file *m, void *v)
 {
-	if (saved_boot_config)
-		seq_puts(m, saved_boot_config);
+	if (saved_boot_config) {
+		char *b = kstrdup(saved_boot_config, GFP_KERNEL);
+
+		if (b) {
+			char *s;
+
+			while ((s = strstr(b, "verifiedbootstate = \"orange\"")))
+				memcpy(s, "verifiedbootstate  = \"green\"", 27);
+			while ((s = strstr(b, "device_state = \"unlocked\"")))
+				memcpy(s, "device_state   = \"locked\"", 25);
+			while ((s = strstr(b, "flash.locked = \"0\"")))
+				memcpy(s, "flash.locked = \"1\"", 18);
+			while ((s = strstr(b, "veritymode = \"logging\"")))
+				memcpy(s, "veritymode = \"enforcing\"", 24);
+			seq_puts(m, b);
+			kfree(b);
+		} else {
+			seq_puts(m, saved_boot_config);
+		}
+	}
+
 	return 0;
 }
 
 static int __init copy_xbc_key_value_list(char *dst, size_t size)
 {
