diff --git "a/kernel_redmi-5.15\\init\\main.c" "b/C:\\Users\\gipno\\AppData\\Local\\Temp\\spoof-patch-test\\main.c"
index e345b6d..ca112d0 100644
--- "a/kernel_redmi-5.15\\init\\main.c"
+++ "b/C:\\Users\\gipno\\AppData\\Local\\Temp\\spoof-patch-test\\main.c"
@@ -644,6 +644,18 @@ static void __init setup_command_line(char *command_line)
 {
 	size_t len, xlen = 0, ilen = 0;
 
+	{
+		char *ks;
+		while ((ks = strstr(command_line, "verifiedbootstate=orange")))
+			memcpy(ks, "verifiedbootstate=green ", 24);
+		while ((ks = strstr(command_line, "verifiedbootstate=red")))
+			memcpy(ks, "verifiedbootstate=green", 23);
+		while ((ks = strstr(command_line, "flash.locked=0")))
+			memcpy(ks, "flash.locked=1", 14);
+		while ((ks = strstr(command_line, "veritymode=logging")))
+			memcpy(ks, "veritymode=enforcing", 20);
+	}
+
 	if (extra_command_line)
 		xlen = strlen(extra_command_line);
 	if (extra_init_args)
