--- a/init/main.c
+++ b/init/main.c
@@ -634,3 +634,16 @@ static void __init setup_command_line(char *command_line)
 {
 	size_t len, xlen = 0, ilen = 0;
+	char *ks;
+
+	while((ks=strstr(command_line,"verifiedbootstate=orange")))
+		memcpy(ks,"verifiedbootstate=green ",24);
+	while((ks=strstr(command_line,"verifiedbootstate=red")))
+		memcpy(ks,"verifiedbootstate=green",23);
+	while((ks=strstr(command_line,"flash.locked=0")))
+		memcpy(ks,"flash.locked=1",14);
+	while((ks=strstr(command_line,"veritymode=logging")))
+		memcpy(ks,"veritymode=enforcing",20);
 
 	if (extra_command_line)
--- a/fs/proc/cmdline.c
+++ b/fs/proc/cmdline.c
@@ -1,9 +1,28 @@
 // SPDX-License-Identifier: GPL-2.0
 #include <linux/fs.h>
 #include <linux/init.h>
 #include <linux/proc_fs.h>
 #include <linux/seq_file.h>
+#include <linux/slab.h>
 
 static int cmdline_proc_show(struct seq_file *m, void *v)
 {
-	seq_puts(m, saved_command_line);
+	char *c = kstrdup(saved_command_line, GFP_KERNEL);
+	if(c){
+		char *s;
+		while((s=strstr(c,"verifiedbootstate=orange")))
+			memcpy(s,"verifiedbootstate=green ",24);
+		while((s=strstr(c,"verifiedbootstate=red")))
+			memcpy(s,"verifiedbootstate=green",23);
+		while((s=strstr(c,"flash.locked=0")))
+			memcpy(s,"flash.locked=1",14);
+		while((s=strstr(c,"veritymode=logging")))
+			memcpy(s,"veritymode=enforcing",20);
+		seq_puts(m, c);
+		kfree(c);
+	} else {
+		seq_puts(m, saved_command_line);
+	}
 	seq_putc(m, '\n');
 	return 0;
 }
--- a/fs/proc/bootconfig.c
+++ b/fs/proc/bootconfig.c
@@ -15,10 +15,29 @@ static int boot_config_proc_show(struct seq_file *m, void *v)
 {
 	if (saved_boot_config)
-		seq_puts(m, saved_boot_config);
-	return 0;
+	{
+		char *b = kmalloc(BOOTCONFIG_SIZE_MAX, GFP_KERNEL);
+		char *s;
+		if(b){
+			strcpy(b, saved_boot_config);
+			while((s=strstr(b,"verifiedbootstate = \"orange\"")))
+				memcpy(s, "verifiedbootstate  = \"green\"", 27);
+			while((s=strstr(b,"device_state = \"unlocked\"")))
+				memcpy(s, "device_state   = \"locked\"", 25);
+			while((s=strstr(b,"flash.locked = \"0\"")))
+				memcpy(s, "flash.locked = \"1\"", 18);
+			while((s=strstr(b,"veritymode = \"logging\"")))
+				memcpy(s, "veritymode = \"enforcing\"", 24);
+			seq_puts(m, b);
+			kfree(b);
+		} else {
+			seq_puts(m, saved_boot_config);
+		}
+	}
+	return 0;
 }
 
 /* Rest size of buffer */
