diff --git a/fs/proc/bootconfig.c b/fs/proc/bootconfig.c
index 2e244ad..d2540ec 100644
--- a/fs/proc/bootconfig.c
+++ b/fs/proc/bootconfig.c
@@ -9,6 +9,7 @@
 #include <linux/seq_file.h>
 #include <linux/bootconfig.h>
 #include <linux/slab.h>
+#include <linux/string.h>
 
 static char *saved_boot_config;
 
@@ -46,7 +47,16 @@ static int __init copy_xbc_key_value_list(char *dst, size_t size)
 		vnode = xbc_node_get_child(leaf);
 		if (vnode) {
 			xbc_array_for_each_value(vnode, val) {
+				if (strstr(key, "verifiedbootstate") && !strcmp(val, "orange"))
+					val = "green";
+				else if (strstr(key, "device_state") && !strcmp(val, "unlocked"))
+					val = "locked";
+				else if (strstr(key, "flash.locked") && !strcmp(val, "0"))
+					val = "1";
+				else if (strstr(key, "veritymode") && !strcmp(val, "logging"))
+					val = "enforcing";
+
 				if (strchr(val, '"'))
 					q = '\'';
 				else
 					q = '"';
diff --git a/fs/proc/cmdline.c b/fs/proc/cmdline.c
index fa762c5..f877bce 100644
--- a/fs/proc/cmdline.c
+++ b/fs/proc/cmdline.c
@@ -3,10 +3,39 @@
 #include <linux/init.h>
 #include <linux/proc_fs.h>
 #include <linux/seq_file.h>
+#include <linux/slab.h>
+#include <linux/string.h>
+
+static void replace_str(char *str, const char *old_str, const char *new_str)
+{
+	char *pos;
+	size_t old_len = strlen(old_str);
+	size_t new_len = strlen(new_str);
+	while ((pos = strstr(str, old_str)) != NULL) {
+		if (new_len == old_len) {
+			memcpy(pos, new_str, new_len);
+		} else {
+			memmove(pos + new_len, pos + old_len, strlen(pos + old_len) + 1);
+			memcpy(pos, new_str, new_len);
+		}
+		str = pos + new_len;
+	}
+}
 
 static int cmdline_proc_show(struct seq_file *m, void *v)
 {
-	seq_puts(m, saved_command_line);
+	char *c = kmalloc(strlen(saved_command_line) + 128, GFP_KERNEL);
+	if (c) {
+		strcpy(c, saved_command_line);
+		replace_str(c, "verifiedbootstate=orange", "verifiedbootstate=green ");
+		replace_str(c, "verifiedbootstate=red", "verifiedbootstate=green");
+		replace_str(c, "flash.locked=0", "flash.locked=1");
+		replace_str(c, "veritymode=logging", "veritymode=enforcing");
+		seq_puts(m, c);
+		kfree(c);
+	} else {
+		seq_puts(m, saved_command_line);
+	}
 	seq_putc(m, '\n');
 	return 0;
 }
diff --git a/init/main.c b/init/main.c
index e345b6d..d47aeda 100644
--- a/init/main.c
+++ b/init/main.c
@@ -643,7 +643,35 @@ static inline void smp_prepare_cpus(unsigned int maxcpus) { }
+static void replace_str(char *str, const char *old_str, const char *new_str)
+{
+	char *pos;
+	size_t old_len = strlen(old_str);
+	size_t new_len = strlen(new_str);
+	while ((pos = strstr(str, old_str)) != NULL) {
+		if (new_len == old_len) {
+			memcpy(pos, new_str, new_len);
+		} else {
+			memmove(pos + new_len, pos + old_len, strlen(pos + old_len) + 1);
+			memcpy(pos, new_str, new_len);
+		}
+		str = pos + new_len;
+	}
+}
+
 static void __init setup_command_line(char *command_line)
 {
 	size_t len, xlen = 0, ilen = 0;
 
+	replace_str(command_line, "verifiedbootstate=orange", "verifiedbootstate=green ");
+	replace_str(command_line, "verifiedbootstate=red", "verifiedbootstate=green");
+	replace_str(command_line, "flash.locked=0", "flash.locked=1");
+	replace_str(command_line, "veritymode=logging", "veritymode=enforcing");
+
+	if (extra_command_line) {
+		replace_str(extra_command_line, "verifiedbootstate=orange", "verifiedbootstate=green ");
+		replace_str(extra_command_line, "verifiedbootstate=red", "verifiedbootstate=green");
+		replace_str(extra_command_line, "flash.locked=0", "flash.locked=1");
+		replace_str(extra_command_line, "veritymode=logging", "veritymode=enforcing");
+	}
+
 	if (extra_command_line)
 		xlen = strlen(extra_command_line);
 	if (extra_init_args)
 		ilen = strlen(extra_init_args) + 4; /* for " -- " */
