--- a/fs/susfs.c
+++ b/fs/susfs.c
@@ -1164,7 +1164,7 @@ static int copy_config_to_buf(const char *config_string, char *buf_ptr, size_t *
-static int copy_config_to_buf(const char *config_string, char *buf_ptr, size_t *copied_size, size_t bufsize) {
+static int __maybe_unused copy_config_to_buf(const char *config_string, char *buf_ptr, size_t *copied_size, size_t bufsize) {
 		int len = 0;
 		while (len < bufsize) {
 			if (config_string[len] == '\0') break;
@@ -1179,7 +1179,7 @@ static int copy_config_to_buf(const char *config_string, char *buf_ptr, size_t *
-		size_t copied_size = 0;
+		size_t __maybe_unused copied_size = 0;
 		char *buf_ptr = NULL;
 		size_t bufsize = PAGE_SIZE;
 		if (!susfs_ino_allow_list) return 0;
    
